.github/workflows/release.yml

```diff
@@ -0,0 +1,192 @@
+          echo "RELEASE_VERSION=$RELEASE_VERSION" >> $GITHUB_ENV
+          echo "RAW_VERSION=$RAW_VERSION" >> $GITHUB_ENV
+          echo "SHORT_SHA=${GITHUB_SHA::8}" >> $GITHUB_ENV
+
+      - name: Install rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+
+      - name: Check tooling
+        shell: bash
+        run: |
+          tar --version
+          shasum --version
+          cargo --version
+          rustc --version
+
+      - name: Build MacOS binary
+        run: |
+          cargo build --locked --profile release
+          ls -al "${{ env.TARGET_DIR }}/diener"
+
+      - name: Check binary
+        env:
+          BINARY: "${{ env.TARGET_DIR }}/diener"
+        run: |
+          file $BINARY
+          $BINARY --version
+          $BINARY --help
+
+      - name: Compress & sha256
+        run: |
+          tar -czf ${{ env.TARGET_DIR }}/diener_macos.tar.gz -C ${{ env.TARGET_DIR }} diener
+          SHA256=$(shasum -a 256 ${{ env.TARGET_DIR }}/diener_macos.tar.gz | awk '{ print $1}' | tee ${{ env.TARGET_DIR }}/diener_macos.tar.gz.sha256)
+          echo SHA256: $SHA256
+          echo "SHA256=$SHA256" >> $GITHUB_ENV
+
+      - name: Upload MacOS artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: macos
+          path: |
+            ${{ env.TARGET_DIR }}/diener_macos.tar.gz
+            ${{ env.TARGET_DIR }}/diener_macos.tar.gz.sha256
+
+      - name: Install tera MacOS binary v0.1.3
+        run: |
+          URL=https://github.com/chevdor/tera-cli/releases/download/v0.1.3/tera-macos-v0.1.3.tar.gz
+          wget $URL
+          tar xvf tera-macos-v0.1.3.tar.gz -C /usr/local/bin
+          tera --version
+
+      # We do that before checking out master (in case we were not in master already)
+      - name: Prepare new Formula
+        env:
+          NAME: Diener
+          DESCRIPTION: "dependency diener is a tool for easily changing Substrate or Polkadot dependency versions"
+          SITE: https://github.com
+          REPO: paritytech/diener
+          SHA256: ${{env.SHA256}}
+          VERSION: ${{env.RAW_VERSION}}
+        run: |
+          tera --version
+          tera --template templates/formula.rb --env-only > $HOME/diener.rb
+          cat $HOME/diener.rb
+
+      - name: Update Homebrew Formula
+        run: |
+          cp -f $HOME/diener.rb Formula/diener.rb
+          git config --global user.name 'CI'
+          git config --global user.email 'foo@example.invalid'
+          git commit Formula/diener.rb -m "build: new homebrew formula for ${{ env.RELEASE_VERSION }}"
+          git push origin HEAD:master
+
+  create_draft:
+    needs: ["linux", "macos"]
+    name: Create Draft
+    runs-on: ubuntu-latest
+    outputs:
+      release_url: ${{ steps.create-release.outputs.html_url }}
+      asset_upload_url: ${{ steps.create-release.outputs.upload_url }}
+    steps:
+      - uses: actions/checkout@v2
```

Comment:
I think v3 is available. Also we are in a trend of currently pinning the gha using a ref. so I would suggest to pin with the ref of whatever is the current v3. Same for the others.

Reply:
pinned all actions